tests/Feature/Http/Controllers/PostReplyControllerTest.php

```diff
@@ -107,4 +107,153 @@ public function it_does_not_store_a_post_reply_when_the_user_is_not_authenticate
 
         $response->assertRedirect(route('home'));
     }
-}
+
+
+    /**
+     * @test
+     */
+    public function a_moderator_can_delete_post_replies()
+    {
+        $moderator = User::factory()->create()->assignRole('moderator');
+        $user = User::factory()->create()->assignRole('user');
+        $post = Post::factory()->create();
+        $postReply = PostReply::factory()->create();
+        $content = $this->faker->sentence;
+
+        $this->from(route('posts.show', $post->id))
+            ->actingAs($user)
+            ->post(route('replies.store', [$post, $postReply]), [
+            'content' => $content,
+        ]);
+        
+        $this->assertDatabaseHas('post_replies', [
+            'content' => $content,
+        ]);
+        
+        $this->actingAs($moderator);
+        $this->delete(route('replies.destroy', $postReply));
+        $this->assertSoftDeleted('post_replies', [
+            'id' => $postReply->id
+        ]);
+        
+    }
+
+
+    /**
+     * @test
+     */
+    public function a_user_can_delete_his_own_post_replies()
+    {
+        $user = User::factory()->create()->assignRole('user');
+        $post = Post::factory()->create();
+        $postReply = PostReply::factory()->create();
+        $content = $this->faker->sentence;
+
+        $this->from(route('posts.show', $post->id))
+            ->actingAs($user)
+            ->post(route('replies.store', [$post, $postReply]), [
+            'content' => $content,
+        ]);
+        
+        $this->assertDatabaseHas('post_replies', [
+            'content' => $content,
+        ]);
+        
+        $this->actingAs($user);
+        $response = $this->delete(route('replies.destroy', $postReply));
+        $response->assertStatus(302);
+        $this->assertSoftDeleted('post_replies', [
+            'id' => $postReply->id
+        ]);
+
+    }
+
+
+     /**
+     * @test
+     */
+    public function a_user_cannot_delete_post_replies_from_other_users()
+    {
+        $user = User::factory()->create()->assignRole('user');
+        $post = Post::factory()->create();
+        $postReply = PostReply::factory()->create();
+        $content = $this->faker->sentence;
+
+        $this->from(route('posts.show', $post->id))
+            ->actingAs($user)
+            ->post(route('replies.store', [$post, $postReply]), [
+            'content' => $content,
+        ]);
+        
+        $this->assertDatabaseHas('post_replies', [
+            'content' => $content,
+        ]);
+        
+        $this->user2 = User::factory()->create()->assignRole('user');
+
+        $this->actingAs($this->user2);
+        $response = $this->delete(route('replies.destroy', $postReply));
+        $response->assertStatus(403);
+        
+        $this->assertDatabaseHas('post_replies', [
```

Comment:
I guess this returns true even if the reply would have been deleted. Check also that deleted_at of this reply is still null.